### PR TITLE
Making stateless tests with timeout less flaky

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

https://s3.amazonaws.com/clickhouse-test-reports/0/cc3c10d89d5c83c7e194db67b3adb617efcb9cfe/stateless_tests__release_.html

I suspect `timeout` can trigger but a query could be sent just before it, so we have `INSERT` after `DROP`.

cc @CurtizJ 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit a3b61ba665af0cae3057e9128c0139c1a23cf1a4 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51774/a3b61ba665af0cae3057e9128c0139c1a23cf1a4/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by antonio2368** [comment]
Seems like other tests using `timeout` suffer from same thing
https://s3.amazonaws.com/clickhouse-test-reports/0/138e28cf6b98697297c35430f877782c437ddbe0/stateless_tests__asan__[3_4].html

I'll try changing them all

File: tests/queries/0_stateless/01320_create_sync_race_condition_zookeeper.sh [hunk 1]
```diff
@@ -12,22 +12,27 @@ $CLICKHOUSE_CLIENT --allow_deprecated_database_ordinary=1 --query "CREATE DATABA
 
 function thread1()
 {
-    while true; do
-        $CLICKHOUSE_CLIENT -n --query "CREATE TABLE test_01320.r (x UInt64) ENGINE = ReplicatedMergeTree('/test/$CLICKHOUSE_TEST_ZOOKEEPER_PREFIX/table', 'r') ORDER BY x;
-        DROP TABLE test_01320.r;" 2>&1 | grep -F "Code:" | grep -v "UNKNOWN_DATABASE"
+    local TIMELIMIT=$((SECONDS+$1))
+    while [ $SECONDS -lt "$TIMELIMIT" ]; do
+        $CLICKHOUSE_CLIENT -n --query "CREATE TABLE test_01320.r (x UInt64) ENGINE = ReplicatedMergeTree('/test/$CLICKHOUSE_TEST_ZOOKEEPER_PREFIX/table', 'r') ORDER BY x; DROP TABLE test_01320.r;"
```
Comment (antonio2368) [inline tests/queries/0_stateless/01320_create_sync_race_condition_zookeeper.sh:17]: @tavplubix FYI I saw you changed this here https://github.com/ClickHouse/ClickHouse/pull/51742
but as it seems like a more widespread problem I removed your changes and applied fix as for everything else

**Review by tavplubix — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests (release) [2/4] - https://github.com/ClickHouse/ClickHouse/issues/51840
Stateless tests (release, wide parts enabled) - 01606_git_import - `unable to access 'https://github.com/ClickHouse/clickhouse-odbc.git/': gnutls_handshake() failed: Error in the pull function`

**Pull request merged**